### PR TITLE
Added Credential Encryption Document - Fixes #150

### LINKED ISSUE #150: Add DSC Credential Encryption guidance document
This was raised by @TravisEz13 in [this DSCResources issue](https://github.com/PowerShell/DscResources/issues/87).

I have completed the document as suggested by @TravisEz13 based on his [blog post](https://blogs.msdn.microsoft.com/powershell/2014/01/31/want-to-secure-credentials-in-windows-powershell-desired-state-configuration/).

It should complete task 1:

> "We should create a doc on best practices for securing the environment. The first version could be based on the blog and should be put in PowerShell-Docs/Dsc (this is the proper location because this is a general DSC best practice) documentation (e.g. https://github.com/PowerShell/PowerShell-Docs/blob/master/dsc/configDataCredentials.md)"

I will submit the PR for this shortly.

### PR DESCRIPTION
**Pull request by PlagueHO**
Adds credentialEncryption.md document based on blog post from @TravisEz13.

I also included some more information on the requirements of the encryption certificate as this hasn't been made available anywhere else I could find (e.g. the EKU required). So I would appreciate if someone at MSFT could verify this information.

Thanks!

### CODE REVIEW

File: dsc/credentialEncryption.md [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+When dealing with encryption of DSC configuration, it is important to understand the DSC **configuration data** structure.
+This structure is used to separate configuration data from the logic of a configuration file.
+It allows node specific configuration data to be passed to a DSC configuration when it is run.
+See [Separating Configuration and Environment Data](configData.md) for more information on the **configuration data** structure. 
+
+The elements that can be configured for each node that are related to credential encryption are:
+* **NodeName** - the name of the target node that the credential encryption is being configured for.
+* **PsDscAllowPlainTextPassword** - whether unencrypted credentials will be allowed to be passed to this node. This is **not recommended**.
+* **Thumbprint** - the thumbprint of the certificate that will be used to decrypt the credentials in the DSC Configuration on the _Target Node_. **This certificate must exist in the Local Machine certificate store on the Target Node.**
+* **CertificateFile** - the certificate file (containing the public key only) that should be used to encrypt the credentials for the _Target Node_.
```
Comment (TravisEz13) [inline dsc/credentialEncryption.md]: There were question on the blog if you could al

File: dsc/credentialEncryption.md [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+$ConfigData = @{ 
+        AllNodes = @(     
+                        @{  
+                            # The name of the node we are describing 
+                            NodeName        = “targetNode”
+
+                            # The path to the .cer file containing the 
+                            # public key of the Encryption Certificate 
+                            # used to encrypt credentials for this node 
+                            CertificateFile = “C:\publicKeys\targetNode.cer”
```
Comment (TravisEz13) [inline dsc/credentialEncryption.md]: There were questions on the blog if you could also use a local cert in the store.  the answer is yes, but you should avoid transporting the private key protected with a well known password.  In most cases where i asked how they would implement this, the answer came down to this implementation.

**Comment by TravisEz13** [comment]
https://github.com/PowerShell/PowerShell-Docs/blob/master/dsc/secureMOF.md already exists and contains most of the information.

**Comment by PlagueHO** [comment]
Yes, you are quite right - the secureMOF.md is almost identical to the new credentialEncryption.md...

I'm wondering then if credentialEncryption.md is even necessary?

**Comment by TravisEz13** [comment]
I don't think it is.  Can you review secureMof and see if anything is missing.  If it is send a PR for it?

**Comment by PlagueHO** [comment]
Sure - so to confirm: I'll update secureMOF.md with any additional info that is in Credential Encryption as well as review any questions you answered on your blog post and add the info from your answers.

I'll get that done over the next couple of days.

**Comment by TravisEz13** [comment]
@PlagueHO  sound god

**Comment by PlagueHO** [comment]
I'm going to close this PR and resubmit a new one with the changes (sorry about the delay).

**Pull request closed without merging**